ports/zlib/vcpkg-cmake-wrapper.cmake

```
@@ -0,0 +1,15 @@
+if(NOT ZLIB_ROOT OR ZLIB_ROOT STREQUAL "${_VCPKG_INSTALLED_DIR}/${VCPKG_TARGET_TRIPLET}")
+    set(ZLIB_ROOT "${_VCPKG_INSTALLED_DIR}/${VCPKG_TARGET_TRIPLET}")
+    find_path(ZLIB_INCLUDE_DIR NAMES zlib.h PATHS "${ZLIB_ROOT}/include" NO_DEFAULT_PATH)
+    find_library(ZLIB_LIBRARY_RELEASE NAMES zlib  z PATHS "${ZLIB_ROOT}/lib" NO_DEFAULT_PATH)
+    find_library(ZLIB_LIBRARY_DEBUG   NAMES zlibd z PATHS "${ZLIB_ROOT}/debug/lib" NO_DEFAULT_PATH)
+    if(NOT ZLIB_INCLUDE_DIR OR NOT ZLIB_LIBRARY_RELEASE OR (NOT ZLIB_LIBRARY_DEBUG AND EXISTS "${ZLIB_ROOT}/debug/lib"))
+        message("Broken installation of vcpkg port zlib")
+    endif()
+    if(CMAKE_VERSION VERSION_LESS 3.4)
+        find_package(SelectLibraryConfigurations)
+        select_library_configurations(ZLIB)
+        unset(ZLIB_FOUND)
+    endif()
+endif()

```

Why so complicated? If the wrapper is installed and called you can automatically assume that you are wanting zlib from vcpkg (because otherwise the wrapper would not be installed). 
```suggestion
set(ZLIB_ROOT "${_VCPKG_INSTALLED_DIR}/${VCPKG_TARGET_TRIPLET}")
find_path(ZLIB_INCLUDE_DIR NAMES zlib.h PATHS "${ZLIB_ROOT}/include" NO_DEFAULT_PATH)
find_library(ZLIB_LIBRARY_RELEASE NAMES zlib  z PATHS "${ZLIB_ROOT}/lib" NO_DEFAULT_PATH)
find_library(ZLIB_LIBRARY_DEBUG   NAMES zlibd z PATHS "${ZLIB_ROOT}/debug/lib" NO_DEFAULT_PATH)
```
the rest should be handled by the `find_package` call (e.g. select_library_configurations and stuff).

---

I really tried to minimize what is called complicated now. The purpose of the code is:
- Ensure that the libraries are *not* looked up in another path if not found in vcpkg. You may perhaps rely on zlib.h, but no so much on the libs: Most platforms come with some version of zlib. The library names vary with the platform. The community triplets are not even tested by CI. If the module does its own search, it may find a libary in a different place, so that a configuration failure would go unnoticed.
- But do not break release-only triplets.
- Ensure that configuration is done correctly even in CMake 3.1 ... 3.3. where the module does not call select_library_configurations. 

Of course, things might be more compact with a standardized `vcpkg_find_libraries` which would encapsulate the vcpkg paths and the checks for existance:

~~~
vcpkg_find_libraries(ZLIB
    HEADER zlib.h
    LIBRARY_NAMES_RELEASE zlib z
    LIBRARY_NAMES_DEBUG zlibd z
) 
~~~

(Leaving the `select_library_configurations`  as proposed.)